### PR TITLE
Add Benin Holidays

### PR DESCRIPTION
**Pull request by Massoud114**
# Contributing a new country ?

* [x] Yes, I've checked, and there is no current PR for Benin. I had already opened one and it was closed.

1. Create a new class in the Countries directory. It should extend the Country class.
2. Add a test for the new country in the tests directory.
3. Run the tests so a snapshot gets created.
4. Verify the result in the newly created snapshot is correct.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,7 +22,8 @@
     "require": {
         "php": "^8.1",
         "nesbot/carbon": "^2.72.1",
-        "ext-calendar": "*"
+        "ext-calendar": "*",
+        "geniusts/hijri-dates": "^1.1"
```
Comment (Nielsvanpach) [inline composer.json:26]: I prefer to not use this package:
https://github.com/spatie/holidays/pull/155#issuecomment-1904467312

You can have a look at Turkey and how they solved Islamic holidays:
https://github.com/spatie/holidays/blob/main/src/Countries/Turkey.php#L11

File: src/Countries/Benin.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            'Jour de Noël' => '12-25',
+            'Jour de l\'assomption' => '08-15',
+        ], $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = CarbonImmutable::createFromTimestamp(easter_date($year))
+            ->setTimezone('Africa/Porto-Novo');
```
Comment (Nielsvanpach) [inline src/Countries/Benin.php:34]:
```suggestion
        $easter = $this->easter($year);
```

File: tests/.pest/snapshots/Countries/BeninTest/it_can_calculate_benin_holidays.snap [hunk 3]
```diff
@@ -0,0 +1,54 @@
+[
+    {
+        "name": "F\u00eate du Ramadan",
```
Comment (Nielsvanpach) [inline tests/.pest/snapshots/Countries/BeninTest/it_can_calculate_benin_holidays.snap:3]: Results seem to be wrong? Have you checked all results in the snapshot?

File: tests/Countries/BeninTest.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Spatie\Holidays\Tests\Countries;
+
+use Carbon\CarbonImmutable;
+use Spatie\Holidays\Holidays;
+
+it('can calculate benin holidays', function () {
+    CarbonImmutable::setTestNowAndTimezone('2024-01-01');
```
Comment (Nielsvanpach) [inline tests/Countries/BeninTest.php:9]:
```suggestion
    CarbonImmutable::setTestNow('2024-01-01');
```

File: composer.json (hunk 1 above)
Comment (Massoud114) [reply composer.json:26]: Can I extend the Turkey class to avoid repeating the long list of Islamic dates?

Comment (Nielsvanpach) [reply composer.json:26]: As far as I understood @thecaliskan, this is different for most countries, so not really copy pastable.

File: tests/.pest/snapshots/Countries/BeninTest/it_can_calculate_benin_holidays.snap (hunk 3 above)
Comment (Massoud114) [reply tests/.pest/snapshots/Countries/BeninTest/it_can_calculate_benin_holidays.snap]: The content of the snapshot is badly encoded for French characters (é,è,ê...), I don't know how to fix it.

**Comment by Nielsvanpach** [comment]
Thanks!

**Pull request merged**